strawberry/tools/create_type.py

```diff
@@ -33,14 +34,15 @@ def create_type(name: str, fields: List[StrawberryField]) -> Type:
                 )
             )
 
-        dataclass_fields.append(
-            (
-                field.graphql_name,
-                field.type,
-                dataclasses_field(default=field),
-            )
-        )
+    for field in fields:
+        if not isinstance(field, StrawberryField):
+            raise TypeError("Field is not an instance of StrawberryField")
+
+        namespace[field.python_name] = field
+        annotations[field.python_name] = field.type
+
+    namespace["__annotations__"] = annotations  # type: ignore
 
-    cls = make_dataclass(name, fields=dataclass_fields)
+    cls = types.new_class(name, (), {}, lambda ns: ns.update(namespace))
```

Comment:
oh, @jkimbo @BryceBeagle I've removed the make_dataclass creation as it was breaking the new code (calling dataclasses twice was removing our custom fields)

Reply:
But won't that mean that the create_type function no longer returns a dataclass?

Also I don't understand why this change means we're calling make_dataclass twice?

Reply:
before this change we are creating the dataclass twice:

1. cls = make_dataclass(name, fields=dataclass_fields)
2. return strawberry.type(cls)

`strawberry.type` runs `dataclasses.dataclass` on cls, effectively creating the dataclass again, that wouldn't be a be problem normally, but it was also overriding the values of the fields.

Doing a check in `strawberry.type` breaks inheritance unfortunately :)

hope this makes sense :)

we still return a dataclass since `strawberry.type` creates one for you :)

Reply:
Ah yeah that makes sense. Good catch!